autogen/gradio_gui/gradio_service.py

```diff
@@ -0,0 +1,449 @@
+import os; os.environ['no_proxy'] = '*' # Avoid unexpected pollution caused by proxy networks
+import pickle
+import importlib
+import base64
+
+def main(plugins):
+    import gradio as gr
+    if gr.__version__ not in ['3.32.6']: 
+        # this is a special version of gradio, which is not available on pypi.org
+        raise ModuleNotFoundError(
+            "Use the built-in Gradio for the best experience!" + 
+            "Please run `pip install -r https://github.com/binary-husky/gpt_academic/raw/master/docs/gradio-3.32.6-py3-none-any.whl` Command to install built-in Gradio and other dependencies, See details in requirements.txt.")
```

Comment:
Is it possible to use the official version of gradio? What's the differences between your version and the official version?

Reply:
I changed the installation source to https://pypi.org/project/gradio-stable-fork/ (pip install gradio-stable-fork). 

The official Gradio has not implemented a floating and draggable secondary window and tooltip menu, (which is implemented in gradio-stable-fork), these functionality are very useful in AutoGen.